src/detect-content.c

```diff
@@ -405,6 +406,12 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
                     cd->content_len, cd->offset, right_edge, max_right_edge);
             return FALSE;
         }
+	if (max_right_edge_i == distance_size) {
```

Comment:
This seems to exact, can you check https://github.com/OISF/suricata-verify/pull/80/files#r298919007?

I think testing distance directly isn't correct as it's distance after a prior pattern match, of which the length should also be considered.